src/core/index.ts

```diff
@@ -57,13 +63,80 @@ export const journey = wrapFnWithLocation(
   }
 );
 
-export const step = wrapFnWithLocation(
+journey.skip = wrapFnWithLocation(
+  (
+    location: Location,
+    options: JourneyOptions | string,
+    callback: JourneyCallback
+  ) => {
+    log(`Journey register: ${JSON.stringify(options)}`);
+    if (typeof options === 'string') {
+      options = { name: options, id: options };
+    }
+    const j = new Journey(options, callback, location);
+    j.skip = true;
+    runner.addJourney(j);
+    return j;
+  }
+);
+
+journey.only = wrapFnWithLocation(
+  (
+    location: Location,
+    options: JourneyOptions | string,
+    callback: JourneyCallback
+  ) => {
+    log(`Journey register: ${JSON.stringify(options)}`);
+    if (typeof options === 'string') {
+      options = { name: options, id: options };
+    }
+    const j = new Journey(options, callback, location);
+    j.only = true;
+    runner.addJourney(j);
+    return j;
+  }
+);
+
+export const step: StepWithAnnotations = wrapFnWithLocation(
   (location: Location, name: string, callback: VoidCallback) => {
     log(`Step register: ${name}`);
     return runner.currentJourney?.addStep(name, callback, location);
   }
 );
 
+step.skip = wrapFnWithLocation(
+  (location: Location, name: string, callback: VoidCallback) => {
+    log(`Step register: ${name}`);
+    return runner.currentJourney?.addStep(name, callback, location, {
+      skip: true,
+    });
+  }
+);
+
+/**
+ * Failure of soft step will not skip rest of the steps
+ */
+step.soft = wrapFnWithLocation(
```

Comment:
Been thinking about this today, I am not convinced that this makes sense. Feels like we are trying to work around the `expect.soft` problem. Lets re-explore this in a separate PR. Mainly suggesting because once we have the expect.soft support this becomes useless. 

If the devs want to complete a step without failing, It would be more ideal to either comment out then failing method `expect.blah` or change that call to a soft assertion. I dont think anyone would have a step just for a assertion and would call `.soft` on that whole step instead.

Reply:
Thanks Shahzad, I am really not convinced on the soft usecase for step. Would you suggest some usecase?

A step can error only if there are any expect statements, otherwise it cant unless its a JS bug which users can figure out if they run locally

if users want to soft assert, they can do expect.soft or so if we cannot add support via the PW expect. Lets do it as followup call, unless there is a need for the step.soft. We would be in a better position to add it later than to deprecate it based on what I said above.